--- a/dts/mt6897.dts
+++ b/dts/mt6897.dts
@@ -3397,11 +3397,18 @@
 
 		gpu1 {
 			polling-delay = <0x00>;
-			polling-delay-passive = <0x00>;
+			polling-delay-passive = <0x3e8>;
 			thermal-sensors = <0x85 0x1f>;
 
 			trips {
 
+				gpu1-throttle {
+					temperature = <0x19a28>;
+					hysteresis = <0x1388>;
+					type = "passive";
+					phandle = <0x384>;
+				};
+
 				gpu1_crit: gpu1-crit {
 					temperature = <0x1d0d8>;
 					hysteresis = <0x7d0>;
@@ -3409,15 +3416,30 @@
 					phandle = <0x22a>;
 				};
 			};
+
+			cooling-maps {
+
+				map0 {
+					trip = <0x384>;
+					cooling-device = <0x2cd 0xffffffff 0xffffffff>;
+				};
+			};
 		};
 
 		gpu2 {
 			polling-delay = <0x00>;
-			polling-delay-passive = <0x00>;
+			polling-delay-passive = <0x3e8>;
 			thermal-sensors = <0x85 0x20>;
 
 			trips {
 
+				gpu2-throttle {
+					temperature = <0x19a28>;
+					hysteresis = <0x1388>;
+					type = "passive";
+					phandle = <0x385>;
+				};
+
 				gpu2_crit: gpu2-crit {
 					temperature = <0x1d0d8>;
 					hysteresis = <0x7d0>;
@@ -3425,6 +3447,14 @@
 					phandle = <0x22b>;
 				};
 			};
+
+			cooling-maps {
+
+				map0 {
+					trip = <0x385>;
+					cooling-device = <0x2cd 0xffffffff 0xffffffff>;
+				};
+			};
 		};
 
 		mtktsAP: mtktsAP {
